### PR TITLE
Issue/7937 jetpack installation tracks

### LINKED ISSUE #7937: [Simplified Login] Tracks for Jetpack installation
Internal discussion: p1669106173886659-slack-C046HDZL87J

##### For the Install or Connect button at the bottom of the page
- login_jetpack_connection_error_shown
- login_jetpack_connect_button_tapped
- login_jetpack_required_screen_viewed
- login_jetpack_setup_button_tapped

##### For the site credentials screen
- login_jetpack_site_credential_screen_viewed
- login_jetpack_site_credential_screen_dismissed
- login_jetpack_site_credential_install_button_tapped
- login_jetpack_site_credential_reset_password_button_tapped
- login_jetpack_site_credential_did_show_error_alert
- login_jetpack_site_credential_did_finish_login

##### For Install Jetpack screen
- login_jetpack_setup_screen_viewed
- login_jetpack_setup_screen_dismissed

- login_jetpack_setup_install_successful
- login_jetpack_setup_install_failed

- login_jetpack_setup_activation_successful
- login_jetpack_setup_activation_failed

- login_jetpack_setup_fetch_jetpack_connection_url_successful
- login_jetpack_setup_fetch_jetpack_connection_url_failed

- login_jetpack_setup_cannot_find_WPCOM_user
- login_jetpack_setup_authorized_using_different_wpcom_account
- login_jetpack_setup_all_steps_marked_done
- login_jetpack_setup_error_checking_jetpack_connection
- login_jetpack_fetching_wpcom_sites_failed

- login_jetpack_setup_go_to_store_button_tapped

##### Update `site_picker_site_discovery` event with the following properties
- has_wordpress
- is_wpcom
- is_jetpack_installed
- is_jetpack_active
- is_jetpack_connected

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #7937 

⚠️ please don't merge before #7936  

### Description
This PR adds tracking for the Jetpack installation flow.

There is still one question about an additional event that we may need to track p1669288236797139/1669106173.886659-slack-C046HDZL87J, @hafizrahman if it gets answered before my morning, please feel free to add it.

### Testing instructions
Go through the steps (Jetpack not installed case) described in #7889, this should result in the following events:

```
Tracked: site_picker_add_a_store_tapped, Properties: {"is_debug":true}
Tracked: site_picker_connect_existing_store_tapped, Properties: {"is_debug":true}
Tracked: site_picker_site_discovery, Properties: {"has_wordpress":true,"is_wpcom":false,"is_jetpack_installed":false,"is_jetpack_active":false,"is_jetpack_connected":false,"is_debug":true}
Tracked: simplified_login_experiment, Properties: {"experiment_variant":"NATIVE", is_debug":true}
Tracked: login_jetpack_required_screen_viewed, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_button_tapped, Properties: {"is_debug":true}
Tracked: login_jetpack_site_credential_screen_viewed, Properties: {"is_debug":true}
Tracked: login_jetpack_site_credential_reset_password_button_tapped, Properties: {"is_debug":true}
Tracked: login_jetpack_site_credential_install_button_tapped, Properties: {"is_debug":true}
Tracked: login_jetpack_site_credential_did_finish_login, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_screen_viewed, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_install_successful, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_activation_successful, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_fetch_jetpack_connection_url_successful, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_all_steps_marked_done, Properties: {"is_debug":true}
Tracked: login_jetpack_setup_go_to_store_button_tapped, Properties: {"is_debug":true}
```

For the error events, testing them will take a lot of effort, I suggest just reviewing the code.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by hafizrahman — APPROVED** [review]
👍🏼 🚢 

~Haven't added the missing ones yet.~ I've added the missing ones.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/jetpack/start/JetpackActivationStartViewModel.kt [hunk 1]
```diff
@@ -37,7 +42,20 @@ class JetpackActivationStartViewModel @Inject constructor(
...
+            stat = if (navArgs.isJetpackInstalled) AnalyticsEvent.LOGIN_JETPACK_CONNECTION_ERROR_SHOWN
+            else AnalyticsEvent.LOGIN_JETPACK_REQUIRED_SCREEN_VIEWED
+        )
+    }
+
     fun onHelpButtonClick() {
+        analyticsTrackerWrapper.track(
+            stat = AnalyticsEvent.LOGIN_JETPACK_SETUP_GET_SUPPORT_BUTTON_TAPPED,
+            properties = mapOf(
+                AnalyticsTracker.KEY_JETPACK_INSTALLATION_STEP to CONNECTION_DISMISSED_KEY
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/jetpack/start/JetpackActivationStartViewModel.kt]: Not sure if CONNECTION_DISMISSED_KEY is the best value for this. My thinking is just to make it easy to find where the help tap comes from.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/jetpack/start/JetpackActivationStartViewModel.kt]: I asked Sharma about it here p1669362602477919/1669106173.886659-slack-C046HDZL87J, we want to be aligned.

**Comment by hichamboushaba** [comment]
I accidentally merged the illustrations PR on this one 🤦

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/jetpack/start/JetpackActivationStartViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/jetpack/start/JetpackActivationStartViewModel.kt]: To close the loop, we went back to just using `connection`.

**Pull request merged**